src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -88,6 +88,7 @@ grpc_status_code TlsFetchKeyMaterials(
   if (credential_reload_config != nullptr) {
     grpc_tls_credential_reload_arg* arg = new grpc_tls_credential_reload_arg();
     arg->key_materials_config = key_materials_config.get();
+    arg->error_details = new grpc_tls_err_details();
```

Comment:
`arg->error_details` and `arg` are always allocated and deallocated together. How about making `arg->error_details` not a pointer but a full object?

Reply:
making a full object would require the `arg` to have a constructor, which is the next step I am going to take. My goal is to make the `arg` a class instead of a struct here.

Reply:
Good point but I don't see that as a problem. Also the class vs struct distinction seems quite irrelevant at this point unless I am missing something.

Reply:
Making `error_details` an object member of `arg` would require `grpc_security.h` to have a new dependency of `grpc_tls_credentials_options.h`, which I think is not feasible here, because `grpc_security.h` is a wrapper interface and shouldn't depend on any other C core header files.
 
IIUC, nearly most of the classes defined here are of the same pointer type, because of the same reason. When defining as pointer type, we could forward declare this class and then use it in  `grpc_security.h`, without need to depend on C core classes, since we don't need to know the how much memory the class that pointer points to is going to take. However, if it's a object, we have to know beforehand, and hence need that dependency.

The main purpose of this PR is to fix that memory leak issue, so I think probably I would try to figure out the solution to this intricate problem later on :-)

Reply:
ok